pkg/buildpipeline/pipeline.go

```diff
@@ -519,6 +574,7 @@ func stageToTask(s Stage, pipelineIdentifier string, buildIdentifier string, nam
 					stepImage = step.Agent.Image
 				}
 				t.Spec.Steps = append(t.Spec.Steps, corev1.Container{
+					// TODO: Step names don't appear to be validated in Build Pipeline _currently_, but should they be?
```

Comment:
Nvm, they need to be RFC 1035 labels as well: https://github.com/knative/build-pipeline/blob/2ce21d9bba67fc3e11122d33bd5b4e1b1a044b9c/pkg/apis/pipeline/v1alpha1/task_validation.go#L73.

Reply:
Makes sense, since each step results in a container.